### PR TITLE
MDEV-12195 - whitespace warnings + printf attributes in sql/log

### PR DESCRIPTION
**Pull request by grooverdan**
No answer on https://lists.launchpad.net/maria-developers/msg10431.html

Fix whitespace.

format attributes highlighted misuse of functions.

### CODE REVIEW

**Comment by svoj** [comment]
Hi Daniel,

Thanks for your contribution. JIRA task has been created to track this pull request: https://jira.mariadb.org/browse/MDEV-12195

This task was added to 10.2 backlog, which is planned to be handled between 2017-03-09 and 2017-03-16.

Thanks,
Sergey

File: sql/log.h [hunk 1]
```diff
@@ -1066,10 +1066,14 @@ int query_error_code(THD *thd, bool not_killed);
...
-void sql_print_error(const char *format, ...);
-void sql_print_warning(const char *format, ...);
-void sql_print_information(const char *format, ...);
-typedef void (*sql_print_message_func)(const char *format, ...);
+void sql_print_error(const char *format, ...)
+                     ATTRIBUTE_FORMAT_FPTR(printf, 1, 2);
+void sql_print_warning(const char *format, ...)
+                       ATTRIBUTE_FORMAT_FPTR(printf, 1, 2);
+void sql_print_information(const char *format, ...)
+                           ATTRIBUTE_FORMAT_FPTR(printf, 1, 2);
```
Comment (svoj) [inline sql/log.h]: The above 3 should be ATTRIBUTE_FORMAT.

Comment (svoj) [inline sql/log.h]: @vuvova pointed out that these functions use my_vsnprintf(), which has extended format. See include/mysql/service_my_snprintf.h for description.

Said the above I believe we can't use ATTRIBUTE_FORMAT() for these functions. :(

File: sql/sql_acl.cc [hunk 2]
```diff
@@ -12273,7 +12273,7 @@ static bool find_mpvio_user(MPVIO_EXT *mpvio)
     DBUG_ASSERT(my_strcasecmp(system_charset_info, mpvio->acl_user->plugin.str,
                               old_password_plugin_name.str));
     my_error(ER_NOT_SUPPORTED_AUTH_MODE, MYF(0));
-    general_log_print(mpvio->thd, COM_CONNECT,
+    general_log_print(mpvio->thd, COM_CONNECT, "%s",
                       ER_THD(mpvio->thd, ER_NOT_SUPPORTED_AUTH_MODE));
```
Comment (svoj) [inline sql/sql_acl.cc]: Thinking more on this: ER_NOT_SUPPORTED_AUTH_MODE is a format string itself. So original code was just fine.

**Comment by svoj** [comment]
Pushed 8c3b98fe5aa8482ffa67e1e38354f5e98113e3ef

I omitted printf attributes and format patch.

**Comment by grooverdan** [comment]
thank you. I will get to looking at feedback.

**Pull request closed without merging**